src/proto/streams/state.rs

```diff
@@ -132,8 +132,10 @@ impl State {
 
     /// Opens the receive-half of the stream when a HEADERS frame is received.
     ///
+    /// is_informational: whether received a 1xx status code
+    ///
     /// Returns true if this transitions the state to Open.
-    pub fn recv_open(&mut self, eos: bool) -> Result<bool, RecvError> {
+    pub fn recv_open(&mut self, eos: bool, is_informational: bool) -> Result<bool, RecvError> {
```

Comment:
I start to worry when methods get multiple boolean arguments, it can be confusing at the call site in my opinion. What if we changed this to pass `&frame::Headers`?